packages/material-ui/src/Autocomplete/Autocomplete.js

```diff
@@ -122,59 +122,59 @@ const AutocompleteRoot = experimentalStyled(
       width: 0,
       minWidth: 30,
     },
-    '&.MuiInput-root': {
-      paddingBottom: 1,
-      '& .MuiInput-input': {
-        padding: '4px 4px 4px 0px',
-      },
+  },
+  '& .MuiInput-root': {
```

Comment:
This is not changing specificity. It is changing the selector entirely.

See https://codesandbox.io/s/css-select-specificity-qq2y9 where `.parent.child` has the same specificity as `.parent .child`. If `.parent.child` would be more specific than `.parent .child` then `.parent.child` would win. But `.parent.child` looses to `.parent .child` because `.parent.child` has the same specificity but is declared before `.parent .child`.

So are we sure we understand this change? Because it's not changing specificity. It just changes the style since one selector is no longer or just now matching.

Reply:
> On the very first demo "Combo box", input element:
> 
> [HEAD](https://next.material-ui.com/components/autocomplete/#combo-box): `.css-125hu5w-MuiAutocomplete-root .MuiAutocomplete-inputRoot.MuiOutlinedInput-root .MuiAutocomplete-input`
> [Specificity](https://specificity.keegan.st/): 4
> 
> [PR](https://deploy-preview-26238--material-ui.netlify.app/components/autocomplete/#combo-box): `.css-168o85v-MuiAutocomplete-root .MuiOutlinedInput-root .MuiAutocomplete-input`
> [Specificity](https://specificity.keegan.st/): 3

These are not the selectors we're talking about here. You're mentioning `.MuiOutlinedInput-root` when I target `.MuiInput-root`

Reply:
> So I think that providing empty space is increasing the specificity instead of decreasing it, right?

I just explained that this is not increasing specificity. Not following my interpretation of [CSS specificity](https://www.w3.org/TR/CSS2/cascade.html#specificity) nor in praxis (see codesandbox). The selectors Olivier posted have different specificity because they have more class selectors, not because they have `.a.b` instead of `.a .b`.

Reply:
Did you un-nest the selector? It's incredibly hard to follow this syntax soup right now.

Reply:
Can we lint against these kind of customizations? The state of the styles right now is just terrible regarding readability. I was hoping this improves with time. But I'm almost certain we have a dozen issues right now similar to this one. At some point there's too much syntax. And the new `experimentalStyled` has exceeded this threshhold. We have to think about long-term maintainability. This will lead to another picker fiasco if we can't keep the API in check.